File: Selenium/Activities/Activity14.java
import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity14 {

    public static void main(String[] args) {

        ChromeDriver driver = new ChromeDriver();
        WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
        
        driver.manage().window().maximize();

        driver.get("https://training-support.net/webelements/tables");
        System.out.println("Title of page is : " + driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
        System.out.println("The number of rows in table is :" +rows.size());
        
        List<WebElement> columns = driver.findElements(By.xpath("//table/thead/tr/th"));
        System.out.println("The number of columns in table is :" +columns.size());
       
        WebElement bookName = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
        System.out.println("The book name in 5th row is: "+bookName.getText());
        
        WebElement priceHeader = driver.findElement(By.xpath("//table/thead/tr/th[5]"));
        priceHeader.click();
        WebElement bookNameAfter = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
        System.out.println("The book name in 5th row after sorting is: "+bookNameAfter.getText());
        
        driver.quit();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait

driver = webdriver.Chrome()
wait = WebDriverWait(driver, 10)

driver.maximize_window()
driver.get("https://training-support.net/webelements/tables")
print("Title of page is :", driver.title)

rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
print("The number of rows in table is :", len(rows))

columns = driver.find_elements(By.XPATH, "//table/thead/tr/th")
print("The number of columns in table is :", len(columns))

book_name = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[2]")
print("The book name in 5th row is:", book_name.text)

price_header = driver.find_element(By.XPATH, "//table/thead/tr/th[5]")
price_header.click()

book_name_after = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[2]")
print("The book name in 5th row after sorting is:", book_name_after.text)

driver.quit()
